### ISSUE TITLE
Getting error when applying plugin -  Configuration with name 'testCompile' not found.

### ISSUE THREAD

**Issue by fluxtah** [xtext/xtext-gradle-plugin#4]
I am getting an issue using the gradle plugin -  Configuration with name 'testCompile' not found.

**Stacktace**
```
org.gradle.api.ProjectConfigurationException: A problem occurred configuring project ':app'.
        at org.gradle.configuration.project.LifecycleProjectEvaluator.addConfigurationFailure(LifecycleProjectEvaluator.java:91)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.notifyAfterEvaluate(LifecycleProjectEvaluator.java:86)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:65)
        at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:504)
        at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:83)
        at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:47)
        at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:35)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:129)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:80)
        at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:33)
        at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:24)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:36)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:47)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:34)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:35)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:24)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.StartStopIfBuildAndStop.execute(StartStopIfBuildAndStop.java:33)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:71)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:69)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:69)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:34)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:70)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:34)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.DaemonHygieneAction.execute(DaemonHygieneAction.java:39)
        at org.gradle.launcher.daemon.server.exec.DaemonCommandExecution.proceed(DaemonCommandExecution.java:119)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:46)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
        at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
Caused by: org.gradle.api.artifacts.UnknownConfigurationException: Configuration with name 'testCompile' not found.
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.createNotFoundException(DefaultConfigurationContainer.java:79)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getByName(DefaultNamedDomainObjectCollection.java:192)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.getByName(DefaultConfigurationContainer.java:69)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.getByName(DefaultConfigurationContainer.java:33)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getAt(DefaultNamedDomainObjectCollection.java:204)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.getAt(DefaultConfigurationContainer.java:33)
        at org.xtext.gradle.XtextPlugin$1.execute(XtextPlugin.java:156)
        at org.xtext.gradle.XtextPlugin$1.execute(XtextPlugin.java:90)
        at org.gradle.listener.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:109)
        at org.gradle.listener.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:98)
        at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:83)
        at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:31)
        at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
        at com.sun.proxy.$Proxy12.afterEvaluate(Unknown Source)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.notifyAfterEvaluate(LifecycleProjectEvaluator.java:79)
        ... 38 more
```

The project is an android gradle project, here are my gradle files.

**/build.gradle
```
buildscript {
    repositories {
        mavenCentral()
        maven {
            url uri('mechanoid-gradle-plugin/repo')
        }
        maven {
            url 'http://dl.bintray.com/ghedeon/gradle'
        }
        maven {
            url "https://oss.sonatype.org/content/repositories/snapshots"
        }
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:1.0.0'
        classpath 'com.robotoworks.mechanoid:mechanoid-gradle-plugin:0.2.2-SNAPSHOT'
        classpath 'com.novoda:gradle-android-command-plugin:1.2.1'
        classpath 'org.robolectric:robolectric-gradle-plugin:0.13.2'
        classpath 'com.novoda:gradle-android-test-plugin:0.10.1'
        classpath 'com.vv.testy:testy:0.1'
        classpath "org.xtext:xtext-gradle-plugin:0.3.6"
    }
}
```

And then we have another sub project

**app/build.gradle**
```
apply plugin: 'com.android.application'
apply plugin: 'mechanoid-gradle-plugin'
apply plugin: 'android-command'
apply plugin: 'testy'
apply plugin: 'org.xtext.xtext'

dependencies {
    // Android dependencies
    compile 'com.android.support:appcompat-v7:21.0.+'
    compile 'com.android.support:cardview-v7:21.0.+'
    compile 'com.android.support:recyclerview-v7:21.0.+'
    compile 'com.android.support:palette-v7:21.0.+'
    compile 'com.google.android.gms:play-services-base:6.5.87'
    compile 'com.google.android.gms:play-services-maps:6.5.87'
    compile 'com.google.android.gms:play-services-appindexing:6.5.87'

    // Repository dependencies
    compile 'com.squareup:otto:1.3.5'
    compile 'javax.inject:javax.inject:1'
    compile 'com.squareup.dagger:dagger:1.2.2'
    provided 'com.squareup.dagger:dagger-compiler:1.2.2'
    compile ('com.github.frankiesardo:icepick:2.0') {
        exclude module: 'support-v4'
    }
    compile 'com.squareup.okhttp:okhttp:2.0.0'
    compile 'com.squareup.okhttp:okhttp-urlconnection:2.0.0'
    compile 'com.squareup.okio:okio:1.0.0'
    compile 'com.squareup.picasso:picasso:2.3.3'
    compile 'com.jakewharton:butterknife:6.1.0'
    compile 'net.hockeyapp.android:HockeySDK:3.0.2'
    compile 'com.sothree.slidinguppanel:library:2.0.2'
    compile 'com.cocosw:undobar:1.5'
    compile 'com.viewpagerindicator:library:2.4.1@aar'
    compile 'uk.co.chrisjenx:calligraphy:1.2.0'
    compile ('com.justeat:com.justeat.mickeydb.lib:1.0.0-SNAPSHOT') {
        exclude module: 'android'
    }

    // Jar libraries
    compile files(
            'libs/mechanoid.jar',
            'libs/libGoogleAnalyticsServices.jar',
            'libs/adxtag3.2.jar')

    provided files(
            'libs/amazon/home-api.jar',
            'libs/amazon/amazonmaps-1.0.2.jar',
            'libs/amazon/motiongestures-api.jar',
            'libs/javawriter-2.1.1.jar')

    // Project dependencies
    compile project(':lib-facebook')

    androidTestCompile ('com.android.support.test.espresso:espresso-core:2.0') {
        exclude group: 'javax.inject'
    }
    androidTestCompile 'com.android.support.test:testing-support-lib:0.1'
    androidTestCompile 'org.mockito:mockito-core:1.9.5'
    androidTestCompile 'com.google.dexmaker:dexmaker:1.2'
    androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'

    xtextTooling 'com.justeat:com.justeat.mickeydb:1.0.0-SNAPSHOT'
}

xtext {
    version = '2.7.3'
    sources {
        srcDir 'src/main/model'
    }

    languages {
        mickey {
            setup = 'com.justeat.mickeydb.MickeyLangStandaloneSetup'
            output.dir = 'src/main/java-gen'
        }
    }
}
```

**Comment by oehme** [comment]
Thanks for the report. I did not anticipate Android usage until now. I think there might be more issues related to that. If you could share a stripped down example project that would be great!

**Comment by fluxtah** [comment]
Thanks for looking! :) I created an example here https://github.com/fluxtah/xtext-android-gradle-example

Let me know if you require further assistance.

Regards,

Ian

**Comment by oehme** [comment]
I gave this some thought and it falls in line with a general refactoring that I plan to do: I want to move the nice Java/Android integration that the Xtend plugin has up to the Xtext plugin. I will do that after our work on the incremental Xtext builder is finished, because that is the time when I need to change this (in a breaking way) anyway.

Until then you can try fixing the symptoms like this:

```gradle
configurations {
  testCompile {
    extendsFrom compile
  }
}
```

**Comment by oehme** [comment]
This is fixed with 1.0.0. We added support for multiple sourceSets and compile classpaths.
